### PR TITLE
Add package: libpsf

### PR DESCRIPTION
**Pull request by healther**

### CODE REVIEW

File: var/spack/repos/builtin/packages/libpsf/package.py [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+    def install(self, spec, prefix):
+        autoreconf = which("autoreconf")
+        libtoolize()
+        autoreconf('--force', '--install', '--verbose', '-Im4')
+        if '+python' in spec:
+            configure("--prefix=%s" % prefix, '--with-python')
+        else:
+            configure("--prefix=%s" % prefix)
+        make()
+        # FIXME: Add logic to build and install here.
```
Comment (davydden) [inline var/spack/repos/builtin/packages/libpsf/package.py]: remove fixme

File: var/spack/repos/builtin/packages/libpsf/package.py [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+from spack import *
+import os
+
+
+class Libpsf(Package):
+    """libpsf is a c++ library that reads Cadence PSF waveform files"""
+
+    homepage = "https://github.com/henjo/libpsf"
+
+    version('master', git='https://github.com/henjo/libpsf.git', branch='master')
```
Comment (davydden) [inline var/spack/repos/builtin/packages/libpsf/package.py]: no releases at all?

File: var/spack/repos/builtin/packages/libpsf/package.py [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+
+    depends_on("autoconf", type="build")
+    depends_on("automake", type="build")
+    depends_on("libtool",  type="build")
+
+    variant('python', default=True, description='Build python wrappers')
+    extends('python',                         when='+python')
+    depends_on('py-setuptools', type='build', when='+python')
+    depends_on('boost+python',                when='+python')
+    depends_on('py-numpy',                    when='+python')
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/libpsf/package.py]: This should probably be `type=('build', 'run')`, right?

File: var/spack/repos/builtin/packages/libpsf/package.py (hunk 2 above)
Comment (citibeth) [inline var/spack/repos/builtin/packages/libpsf/package.py]: Current Spack standard is this version should be called `develop`.

Comment (adamjstewart) [inline var/spack/repos/builtin/packages/libpsf/package.py]: I disagree with that standard. There are no releases, so it won't interfere with version comparison. And a develop version that points to a master branch is misleading. What if someone also adds a develop branch in addition to master?

**Comment by adamjstewart** [comment]
There are dozens of packages currently in Spack with no checksummed versions.

File: var/spack/repos/builtin/packages/libpsf/package.py (hunk 2 above)
Comment (citibeth) [inline var/spack/repos/builtin/packages/libpsf/package.py]: For better or for worse, we have a system of versions where the special version "`develop`" means "the latest", and sorts to be newer than any other versions.  Until we have a more flexible version system available, things will work best if the "latest" version is called `develop`.

**Comment by citibeth** [comment]
> There are dozens of packages currently in Spack with no checksummed versions.

Please see #3112 on why this is a serious problem.  Making a Spack-only release is pretty easy --- and it also aids in reproducability.  If that is just not possible, then I would encourage that this package stays on a fork until it is ready for public consumption.

**Comment by healther** [comment]
I implemented your comments and will check regarding the versioning, until then I close this PR

**Comment by tgamblin** [comment]
@citibeth: I think it's good to encourage these guys to do a release, and we have talked about implementing preferences so that users can decide what they're willing to install (checksummed, https, etc) in #3112.  However, I think we should wait until something like that is implemented before scaring away potential contributors. 

We don't know how the people who use this package typically check it out, and they may be fine with a dev version. I would obviously feel better if we had a security preference, but I don't think enforcing things extremely strictly *right now* adds much. In general, I think we should wait until there is a feature in place to enforce the policy it's meant for.

**Comment by tgamblin** [comment]
@healther: keep us posted on the release and maybe tell us a bit about how you use this package. We'd be interested to know more about your spack use case.

**Comment by citibeth** [comment]
@healther Sorry, I did not mean to discourage you here.  All that is needed is the following added to this PR:
```
# No official releases of libpsf yet exist.  It is recommended
# that users use the latest @develop version.
version('0.0.0.1', git='https://github.com/henjo/libpsf.git',
    commit='6efc14f7c5fa7e09a07e354cc54b9135ec353d70')
```
> We don't know how the people who use this package typically check it out, and they may be fine with a dev version.

I'm assuming that's the case.  We just need something that prevents users from unwittingly installing a dev verison.

**Comment by healther** [comment]
I'm not particularly discouraged, sorry if you got that impression. I totally understand your concern and I don't think it's a good idea to break policy just to keep people interested in contributing. It is a valid (and appreciated) reaction, if you say: Without a checksummed release we don't want it in the official spack repository, please put it in a private additional one.

I'll continue to put additional packages that we have locally upstream. If you don't think they are useful in the general repository please feel free to just state that, no further explanation necessary. As long as I can continue to pester you with git- and spack-related problems I'm happy :)

File: var/spack/repos/builtin/packages/libpsf/package.py [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+    def install(self, spec, prefix):
+        autoreconf = which("autoreconf")
+        libtoolize()
+        autoreconf('--force', '--install', '--verbose', '-Im4')
+        if '+python' in spec:
+            configure("--prefix=%s" % prefix, '--with-python')
+        else:
+            configure("--prefix=%s" % prefix)
+        make()
+        make("install")
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/libpsf/package.py]: If you convert this to `AutotoolsPackage`, you can replace this entire install method with:
```python
def configure_args(self):
    spec = self.spec
    args = []
    if '+python' in spec:
        args.append('--with-python')
    else:
        args.append('--without-python')
    return args
```
Just confirm that the default autoreconf method works.

File: var/spack/repos/builtin/packages/libpsf/package.py [hunk 5]
```diff
@@ -0,0 +1,61 @@
...
+
+    version('develop', git='https://github.com/henjo/libpsf.git', branch='master')
+    # No official releases of libpsf yet exist. It is recommended
+    # that users use the latest @develop version.
+    version('0.0.0.1', git='https://github.com/henjo/libpsf.git',
+        commit='6efc14f7c5fa7e09a07e354cc54b9135ec353d70')
+
+    depends_on("autoconf", type="build")
+    depends_on("automake", type="build")
+    depends_on("libtool",  type="build")
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/libpsf/package.py]: Shouldn't this also depend on `m4`?

**Comment by healther** [comment]
I'm unable to build it locally at the moment and don't have time to investigate -> close and we will reopen if someone fixed it on our end.

**Pull request closed without merging**